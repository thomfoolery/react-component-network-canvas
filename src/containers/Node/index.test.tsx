import React from "react";
import renderer from "react-test-renderer";
import {
  mockUseDragManager,
  mockUseGraphManager,
  mockUseWorkspace,
  mockUseOptions,
  mockUseBridge,
} from "@component/utils/mocks";
import {
  useDragManager,
  useGraphManager,
  useWorkspace,
  useOptions,
  useBridge,
} from "@component/hooks";

import {Node} from "./index";

jest.mock("@component/hooks", () => ({
  ...jest.requireActual("@component/hooks"),
  useDragManager: jest.fn(),
  useGraphManager: jest.fn(),
  useWorkspace: jest.fn(),
  useOptions: jest.fn(),
  useBridge: jest.fn(),
}));

useDragManager.mockImplementation(mockUseDragManager());
useGraphManager.mockImplementation(mockUseGraphManager());
useWorkspace.mockImplementation(mockUseWorkspace());
useOptions.mockImplementation(mockUseOptions());
useBridge.mockImplementation(mockUseBridge());

describe("Node", () => {
  it("renders correctly", () => {
    const key = "node-1";
    const node = {
      id: "node-1",
      position: {
        x: 0,
        y: 0,
      },
      inputPorts: [{id: "node-1-input-1"}],
      outputPorts: [{id: "node-1-output-1"}],
    };
    const tree = renderer.create(<Node key={key} node={node} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
