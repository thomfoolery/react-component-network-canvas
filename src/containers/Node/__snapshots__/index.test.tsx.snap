// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node renders correctly 1`] = `
<div
  className="Node"
  id="Node-node-1"
  style={
    Object {
      "left": 0,
      "top": 0,
    }
  }
>
  <div
    className="NodeInputPorts"
  >
    <div
      className="Port Input"
      data-testid="node-1-input-1"
      id="Port-node-1-input-1"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    />
  </div>
  <div
    className="NodeBody"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  />
  <div
    className="NodeOutputPorts"
  >
    <div
      className="Port Output"
      data-testid="node-1-output-1"
      id="Port-node-1-output-1"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    />
  </div>
</div>
`;
